operations/rack_resiliency/manage_critical_services.md

```diff
@@ -0,0 +1,394 @@
+  critical-service-config.json: |-
+    {
+      "critical_services": {
+        "cilium-operator": {
+          "namespace": "kube-system",
+          "type": "Deployment"
+        },
+        "coredns": {
+          "namespace": "kube-system",
+          "type": "Deployment"
+        },
+        "cray-activemq-artemis-operator-controller-manager": {
+          "namespace": "dvs",
+          "type": "Deployment"
+        },
+        "cray-capmc": {
+          "namespace": "services",
+          "type": "Deployment"
+        },
+        "cray-ceph-csi-cephfs-provisioner": {
+          "namespace": "ceph-cephfs",
+          "type": "Deployment"
+        },
+        "cray-ceph-csi-rbd-provisioner": {
+          "namespace": "ceph-rbd",
+          "type": "Deployment"
+        },
+    ...
+```
+
+Look for the service you want to delete from the configmap inside critical-service-config.json under data section.
+To remove a service:
+    Locate the service entry you want to remove
+    Delete the entire service block, including the trailing comma
+    Ensure the JSON remains valid
+
+For example Removing a service:<br> Before deleting a service:
+
+```json
+{
+    "critical_services": {
+        "cilium-operator": {
+            "namespace": "kube-system",
+            "type": "Deployment"
+        },
+        "coredns": {
+            "namespace": "kube-system",
+            "type": "Deployment"
+        },
+        "cray-ceph-csi-rbd-provisioner": {
+            "namespace": "ceph-rbd",
+            "type": "Deployment"
+        },
+    }
+}
+```
+
+Let's assume, you want to delete coredns then remove all the 3 lines:
+
+```json
+"coredns": {
+    "namespace": "kube-system",
+    "type": "Deployment"
+},
+```
+
+After deleting the lsit will look like:
+
+```json
+{
+    "critical_services": {
+        "cilium-operator": {
+            "namespace": "kube-system",
+            "type": "Deployment"
+        },
+        "cray-ceph-csi-rbd-provisioner": {
+            "namespace": "ceph-rbd",
+            "type": "Deployment"
+        },
+    }
+}
+```
+
+**Important Notes**
+    Always maintain valid JSON syntax
+    Ensure there's no trailing comma after the last service entry
+    Save and exit the editor to apply changes
+    The changes take effect immediately - no restart required
+    Procedure to modify a criticalservice from the configmap
+    Verify the ConfigMap exists:: kubectl get configmaps -n rack-resiliency
+
+**Example:**
+
+```bash
+kubectl get configmaps -n rack-resiliency 
+NAME                 DATA   AGE
+istio-ca-root-cert   1      6d21h
+kube-root-ca.crt     1      6d21h
+rrs-mon-dynamic      2      3d4h
+rrs-mon-static       11     3d4h
+```
+
+Our list of criticalservices is present in rrs-mon-static.
+
+To verify if the list is present run: 
+
+```bash
+kubectl describe configmap rrs-mon-static -n rack-resiliency
+```
+
+#### Example
+
+```bash
+kubectl describe configmap rrs-mon-static -n rack-resiliency 
```

Comment:
Same here, either switch it to `text`, or split it between `bash` for the command, and `test` for the output